tools/run_all_tests.sh

```diff
@@ -20,6 +20,41 @@ COMPONENTS="utils posh binding_c"
 GTEST_FILTER="*"
 BASE_DIR=$PWD
 GCOV_SCOPE="all"
+CONTINUE_ON_ERROR=false
+
+set_sanitizer_options () {
+    # This script runs from build folder
+    cd ..
+    local PROJECT_ROOT=$PWD
+    cd build
+
+    echo "Project root is PROJECT_ROOT"
+
+# new_delete_type_mismatch is disabled because of the below error
+# ==112203==ERROR: AddressSanitizer: new-delete-type-mismatch on 0x622000021100 in thread T0:
+#   object passed to delete has wrong type:
+#   size of the allocated type:   5120 bytes;
+#   size of the deallocated type: 496 bytes.
+#     #0 0x7fd36deac9d8 in operator delete(void*, unsigned long) (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xe19d8)
+#     #1 0x55c8284bcc43 in ReceiverPort_test::~ReceiverPort_test() /home/pbt2kor/data/aos/repos/iceoryx_oss/iceoryx/iceoryx_posh/test/moduletests/test_posh_receiverport.cpp:49
+#     #2 0x55c8284c15d1 in ReceiverPort_test_newdata_Test::~ReceiverPort_test_newdata_Test() /home/pbt2kor/data/aos/repos/iceoryx_oss/iceoryx/iceoryx_posh/test/moduletests/test_posh_receiverport.cpp:137
+#     #3 0x55c8284c15ed in ReceiverPort_test_newdata_Test::~ReceiverPort_test_newdata_Test() /home/pbt2kor/data/aos/repos/iceoryx_oss/iceoryx/iceoryx_posh/test/moduletests/test_posh_receiverport.cpp:137
+#     #4 0x55c82857b2fb in testing::Test::DeleteSelf_() (/home/pbt2kor/data/aos/repos/iceoryx_oss/iceoryx/build/posh/test/posh_moduletests+0x3432fb)
+    echo "OSTYPE is $OSTYPE"
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        ASAN_OPTIONS=detect_leaks=1
+    else
+    # other OS (Mac here)
+    # ==23449==AddressSanitizer: detect_leaks is not supported on this platform.
+        ASAN_OPTIONS=detect_leaks=0
```

Comment:
detect_leaks option wont work in github mac instance ([Mac OS X 10.15.7](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md))
However the [document](https://clang.llvm.org/docs/AddressSanitizer.html#memory-leak-detection) reads as below

> The leak detection is turned on by default on Linux, and can be enabled using ASAN_OPTIONS=detect_leaks=1 on macOS; however, it is not yet supported on other platforms.

Not sure whats the problem here!

Alternate could be to try [standalone](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer#stand-alone-mode) mode

Reply:
I guess this would be okay because LeakSanitizer runs in Linux build